doc/man3/EVP_PKEY_fromdata.pod

```diff
@@ -148,24 +143,42 @@ TODO Write a set of cookbook documents and link to them.
      0x2f, 0x5c, 0xcd, 0x86, 0x71, 0xa8, 0xbf, 0x1a,
      0x47
  };
- const OSSL_PARAM params[] = {
-     OSSL_PARAM_utf8_string("curve-name", "prime256v1"),
-     OSSL_PARAM_BN("priv", priv, sizeof(priv)),
-     OSSL_PARAM_BN("pub", pub, sizeof(pub)),
-     OSSL_PARAM_END
- };
 
  int main()
  {
-     EVP_PKEY_CTX *ctx = EVP_PKEY_CTX_new_from_name(NULL, "EC", NULL);
+     EVP_PKEY_CTX *ctx;
```

Comment:
There a quite a lot of issues in the example..
Here is a working example that I compiled and tested.
(I use OSSL_PKEY_PARAM_GROUP_NAME as the string changed).
```
int main(void)
{
    EVP_PKEY_CTX *ctx = NULL;
    EVP_PKEY *pkey = NULL;
    BIGNUM *priv = NULL;
    OSSL_PARAM_BLD *param_bld = NULL;
    OSSL_PARAM *params = NULL;
    int ret = 0;
    const char *curve_name = "prime256v1";
    /* UNCOMPRESSED FORMAT */
    static const unsigned char pub_data[] = {
       POINT_CONVERSION_UNCOMPRESSED,
       0x1b, 0x93, 0x67, 0x55, 0x1c, 0x55, 0x9f, 0x63,
       0xd1, 0x22, 0xa4, 0xd8, 0xd1, 0x0a, 0x60, 0x6d,
       0x02, 0xa5, 0x77, 0x57, 0xc8, 0xa3, 0x47, 0x73,
       0x3a, 0x6a, 0x08, 0x28, 0x39, 0xbd, 0xc9, 0xd2,
       0x80, 0xec, 0xe9, 0xa7, 0x08, 0x29, 0x71, 0x2f,
       0xc9, 0x56, 0x82, 0xee, 0x9a, 0x85, 0x0f, 0x6d,
       0x7f, 0x59, 0x5f, 0x8c, 0xd1, 0x96, 0x0b, 0xdf,
       0x29, 0x3e, 0x49, 0x07, 0x88, 0x3f, 0x9a, 0x29
    };
    static const unsigned char priv_data[] = {
        0x33, 0xd0, 0x43, 0x83, 0xa9, 0x89, 0x56, 0x03,
        0xd2, 0xd7, 0xfe, 0x6b, 0x01, 0x6f, 0xe4, 0x59,
        0xcc, 0x0d, 0x9a, 0x24, 0x6c, 0x86, 0x1b, 0x2e,
        0xdc, 0x4b, 0x4d, 0x35, 0x43, 0xe1, 0x1b, 0xad
    };

     priv = BN_bin2bn(priv_data, sizeof(priv_data), NULL);

     param_bld = OSSL_PARAM_BLD_new();
     if (priv != NULL && param_bld != NULL) {
         if (OSSL_PARAM_BLD_push_utf8_string(param_bld,
                                             OSSL_PKEY_PARAM_GROUP_NAME,
                                             curve_name, 0)
             && OSSL_PARAM_BLD_push_BN(param_bld, OSSL_PKEY_PARAM_PRIV_KEY, priv)
             && OSSL_PARAM_BLD_push_octet_string(param_bld,
                                                 OSSL_PKEY_PARAM_PUB_KEY,
                                                 pub_data, sizeof(pub_data))) {
                 params = OSSL_PARAM_BLD_to_param(param_bld);
         }
     }

     ctx = EVP_PKEY_CTX_new_from_name(NULL, "EC", NULL);
     if (ctx == NULL
         || params == NULL
         || EVP_PKEY_key_fromdata_init(ctx) <= 0
         || EVP_PKEY_fromdata(ctx, &pkey, params) <= 0) {
         ret = 0;
     } else {
         ret = 1;
         /* Do what you want with |pkey| */
     }

     BN_free(priv);
     OSSL_PARAM_BLD_free_params(params);
     OSSL_PARAM_BLD_free(param_bld);
     EVP_PKEY_free(pkey);
     EVP_PKEY_CTX_free(ctx);

    return ret;
}
```

Reply:
Side note: for a returned number from `main()`, you've got it mixed up...  `main()` is supposed to return an exit code.

Reply:
That is because my test was not done as a main().
It doesnt need to be a main() it could just be a function e.g: int do_key(void)

Reply:
I'm really not very keen on fleshing out example code too much, as the actual thing we want to give an example of *drowns* in code.  That being said, adding a few checks didn't make it much worse, so thank you.